File: municipal-services/ws-services/src/main/java/org/egov/waterconnection/web/models/BillReportData.java
package org.egov.waterconnection.web.models;

import java.math.BigDecimal;
import java.util.Map;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class BillReportData {
	
	@JsonProperty("tenantName")
	private String tenantName = null;
	
	@JsonProperty("consumerName")
	private String consumerName = null;
	
	@JsonProperty("connectionNo")
	private String connectionNo = null;

	@JsonProperty("oldConnectionNo")
	private String oldConnectionNo = null;

	@JsonProperty("consumerCreatedOnDate")
	private String consumerCreatedOnDate = null;
	
	@JsonProperty("penalty")
	private BigDecimal penalty = null;
	
	@JsonProperty("advance")
	private BigDecimal advance = null;
	
	@JsonProperty("demandAmount")
	private BigDecimal demandAmount =null;
	
	@JsonProperty("userId")
	private String userId = null;
}

File: frontend/mgramseva/lib/model/reports/bill_report_data.dart
import 'package:json_annotation/json_annotation.dart';

part 'bill_report_data.g.dart';

@JsonSerializable()
class BillReportData {
  @JsonKey(name: "tenantName")
  String? tenantName;
  @JsonKey(name: "consumerName")
  String? consumerName;
  @JsonKey(name: "connectionNo")
  String? connectionNo;
  @JsonKey(name: "oldConnectionNo")
  String? oldConnectionNo;
  @JsonKey(name: "consumerCreatedOnDate")
  String? consumerCreatedOnDate;
  @JsonKey(name: "penalty")
  double? penalty;
  @JsonKey(name: "advance")
  double? advance;
  @JsonKey(name: "demandAmount")
  double? demandAmount;
  @JsonKey(name: "userId")
  String? userId;

  @override
  String toString() {
    return 'BillReportData{tenantName: $tenantName, consumerName: $consumerName, connectionNo: $connectionNo, oldConnectionNo: $oldConnectionNo, consumerCreatedOnDate: $consumerCreatedOnDate, advance: $advance, penalty: $penalty, demandAmount: $demandAmount, userId: $userId}';
  }

  BillReportData();

  factory BillReportData.fromJson(Map<String, dynamic> json) =>
      _$BillReportDataFromJson(json);

  Map<String, dynamic> toJson(List list) => _$BillReportDataToJson(this);
}
